### PR TITLE
Mark Workload Identity as GA fully.

### PR DESCRIPTION
**Pull request by upodroid**
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/5916

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
container: marked `workload_metadata_config` as GA in `google_container_node_pool`
```

### CODE REVIEW

File: third_party/terraform/resources/resource_container_node_pool.go.erb [hunk 1]
```diff
@@ -772,7 +772,7 @@ func nodePoolUpdate(d *schema.ResourceData, meta interface{}, nodePoolInfo *Node
 			log.Printf("[INFO] Updated workload_metadata_config for node pool %s", name)
 		}
 
-<% end -%>
+
```
Comment (megan07) [inline third_party/terraform/resources/resource_container_node_pool.go.erb:775]: We can probably clean up the extra whitespace here.

File: third_party/terraform/utils/node_config.go.erb [hunk 2]
```diff
@@ -451,7 +449,6 @@ func expandNodeConfig(v interface{}) *containerBeta.NodeConfig {
 	return nc
 }
 
-<% unless version == 'ga' -%>
 func expandWorkloadMetadataConfig(v interface{}) *containerBeta.WorkloadMetadataConfig {
```
Comment (megan07) [inline third_party/terraform/utils/node_config.go.erb:452]: I think we need to address the flatten methods for `WorkloadMetadataConfig` as well.

File: third_party/terraform/website/docs/r/container_cluster.html.markdown [hunk 3]
```diff
@@ -308,7 +308,7 @@ subnetwork in which the cluster's instances are launched.
     Vertical Pod Autoscaling automatically adjusts the resources of pods controlled by it.
     Structure is documented below.
 
-* `workload_identity_config` - (Optional, [Beta](https://terraform.io/docs/providers/google/guides/provider_versions.html))
+* `workload_identity_config` - (Optional)
```
Comment (megan07) [inline third_party/terraform/website/docs/r/container_cluster.html.markdown:311]: Should we remove the `Beta` portion from `workload_metadata_config` as well?

**Review by megan07 — CHANGES_REQUESTED** [review]
Hi @upodroid! Thank you for contributing! With the suggested changes I will run the tests in GA and we can get this merged. As you're making the changes, would you keep an eye out for excess white space and clean that up as well?  Thanks again!

**Comment by upodroid** [comment]
I fixed the issues. Slow test

```
 REDACTED  MCW0CDP3YY  ~  go  …  github.com  hashicorp  terraform-provider-google   master  31⬇  7✎  USAGE  $    make testacc TEST=./google TESTARGS='-run=TestAccContainerNodePool_withWorkloadIdentityConfig'
==> Checking source code against gofmt...
==> Checking that code complies with gofmt requirements...
go generate  ./...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google -v -run=TestAccContainerNodePool_withWorkloadIdentityConfig -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google/version.ProviderVersion=acc"
=== RUN   TestAccContainerNodePool_withWorkloadIdentityConfig
=== PAUSE TestAccContainerNodePool_withWorkloadIdentityConfig
=== CONT  TestAccContainerNodePool_withWorkloadIdentityConfig
--- PASS: TestAccContainerNodePool_withWorkloadIdentityConfig (1675.70s)
PASS
ok      github.com/hashicorp/terraform-provider-google/google   1677.993s
```

**Review by megan07 — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**